### ISSUE TITLE
`jax.jit` does not work with Binning corrections

### ISSUE THREAD

**Issue by eguiraud** [eguiraud/correctionlib-gradients#42]
That's because evaluating a `scipy.interpolate.CubicSpline` is not jax-traceable.

For 1D histograms we might be able to use https://github.com/gradhep/relaxed instead, whose `hist` call I think is jax-traceable.

For multi-dimensional histograms I don't know of a jax-friendly implementation of a bin look-up differentiable relaxation.

**Comment by eguiraud** [comment]
Some discussion at https://iris-hep.slack.com/archives/C0155BGPGE4/p1699120426133849

**Comment by eguiraud** [comment]
options I see:
- re-implement scipy.interpolate.CubicSpline using XLA primitives or jax.numpy methods
- switch to sigmoids as per Lukas' suggestion
- switch to a sum of gaussians
